staging/src/k8s.io/kubectl/pkg/cmd/create/create_priorityclass.go

```diff
@@ -62,50 +92,110 @@ func NewCmdCreatePriorityClass(f cmdutil.Factory, ioStreams genericclioptions.IO
+	cmd.Flags().StringVar(&o.Description, "description", o.Description, i18n.T("description is an arbitrary string that usually provides guidelines on when this priority class should be used."))
+	cmd.Flags().StringVar(&o.PreemptionPolicy, "preemption-policy", o.PreemptionPolicy, i18n.T("preemption-policy is the policy for preempting pods with lower priority."))
+	cmdutil.AddFieldManagerFlagVar(cmd, &o.FieldManager, "kubectl-create")
 	return cmd
 }
 
 // Complete completes all the required options
-func (o *PriorityClassOpts) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
-	name, err := NameFromCommandArgs(cmd, args)
+func (o *PriorityClassOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	var err error
+	o.Name, err = NameFromCommandArgs(cmd, args)
+	if err != nil {
+		return err
+	}
+
+	restConfig, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	o.Client, err = schedulingclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	o.CreateAnnotation = cmdutil.GetFlagBool(cmd, cmdutil.ApplyAnnotationsFlag)
+
+	o.DryRunStrategy, err = cmdutil.GetDryRunStrategy(cmd)
+	if err != nil {
+		return err
+	}
+	dynamicClient, err := f.DynamicClient()
+	if err != nil {
+		return err
+	}
+	discoveryClient, err := f.ToDiscoveryClient()
+	if err != nil {
+		return err
+	}
+	o.DryRunVerifier = resource.NewDryRunVerifier(dynamicClient, discoveryClient)
+	cmdutil.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+	o.PrintObj = func(obj runtime.Object) error {
+		return printer.PrintObj(obj, o.Out)
+	}
+
+	return nil
+}
+
+// Run calls the CreateSubcommandOptions.Run in the PriorityClassOptions instance
+func (o *PriorityClassOptions) Run() error {
+	priorityClass, err := o.createPriorityClass()
 	if err != nil {
 		return err
 	}
 
-	var generator generate.StructuredGenerator
-	switch generatorName := cmdutil.GetFlagString(cmd, "generator"); generatorName {
-	case generateversioned.PriorityClassV1GeneratorName:
-		generator = &generateversioned.PriorityClassV1Generator{
-			Name:             name,
-			Value:            cmdutil.GetFlagInt32(cmd, "value"),
-			GlobalDefault:    cmdutil.GetFlagBool(cmd, "global-default"),
-			Description:      cmdutil.GetFlagString(cmd, "description"),
-			PreemptionPolicy: apiv1.PreemptionPolicy(cmdutil.GetFlagString(cmd, "preemption-policy")),
+	if err := util.CreateOrUpdateAnnotation(o.CreateAnnotation, priorityClass, scheme.DefaultJSONEncoder()); err != nil {
+		return err
+	}
+
+	if o.DryRunStrategy != cmdutil.DryRunClient {
+		createOptions := metav1.CreateOptions{}
+		if o.FieldManager != "" {
+			createOptions.FieldManager = o.FieldManager
+		}
+		if o.DryRunStrategy == cmdutil.DryRunServer {
+			if err := o.DryRunVerifier.HasSupport(priorityClass.GroupVersionKind()); err != nil {
```

Comment:
This is having problems b/c you miss setting that `TypeMeta` below, which ends up kubectl asking the server for support for non-existent GVK. When you add that, you'll notice that these two lines are passing https://github.com/kubernetes/kubernetes/blob/396b90f06ca29c754a81eb6c03cff50b7d9bf265/test/cmd/core.sh#L292-L293

It's best to look through logs looking for last logs with line numbers and try to figure out which might have failed. In your case you'll notice:
```
core.sh:290: Successful get priorityclasses {{range.items}}{{ if eq .metadata.name \"test-priorityclass\" }}found{{end}}{{end}}:: :
```
so the next lines of actual code are those create lines I've pointed out earlier. Running locally both will reveal the problem is with the next and from here it's pretty easy to nail down the actual code issue ;)

Reply:
Great! Thank you for taking the time to walk me through it @soltysh 🙂
Next time I should be more comfortable.